flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/source/assigners/splitters/SampleBucketSplitStrategy.java

```diff
@@ -66,9 +68,13 @@
 @Internal
 public class SampleBucketSplitStrategy implements SplitStrategy {
 
+    private static final Logger LOG = LoggerFactory.getLogger(SampleBucketSplitStrategy.class);
+
     public static final SampleBucketSplitStrategy INSTANCE = new SampleBucketSplitStrategy();
     private static final int DEFAULT_SAMPLING_THRESHOLD = 102400;
     private static final double DEFAULT_SAMPLING_RATE = 0.05;
+    private static final int SAMPLING_COUNT_BY_RATE_THRESHOLD = 1000000;
+    private static final int MAX_SAMPLE_PER_CHUNK = 20;
```

Comment:
Thanks @vanliu-tx for the contribution.
I think we can add a config option `scan.incremental.snapshot.chunk.samples`: The number of samples to take per chunk. 

Refers to https://github.com/apache/flink-connector-mongodb/blob/main/flink-connector-mongodb/src/main/java/org/apache/flink/connector/mongodb/source/enumerator/splitter/MongoSampleSplitter.java.

Reply:
Config option added. Please help to review ;-)